discovery-provider/src/api/v1/helpers.py

```diff
@@ -341,15 +343,52 @@ def extend_reaction(reaction: ReactionResponse):
     return new_reaction
 
 
-def extend_tip(tip):
-    new_tip = tip.copy()
-    new_tip["amount"] = to_wei_string(tip["amount"])
-    new_tip["sender"] = extend_user(tip["sender"])
-    new_tip["receiver"] = extend_user(tip["receiver"])
-    new_tip["followee_supporters"] = [
-        {"user_id": encode_int_id(id)} for id in new_tip["followee_supporters"]
-    ]
-    return new_tip
+class ExtendedTipResult(TypedDict):
+    amount: int
+    sender: User
+    receiver: User
+    slot: int
+    created_at: datetime
+    followee_supporters: List[Dict[str, int]]
+    tx_signature: str
+
+
+def extend_tip(tip: TipResult) -> ExtendedTipResult:
```

Comment:
Previously this was untyped and the types were actually slightly wrong!

Reply:
How were they wrong 🤯